sdk/azidentity/device_code_credential.go

```diff
@@ -15,27 +16,47 @@ const (
 	deviceCodeGrantType = "urn:ietf:params:oauth:grant-type:device_code"
 )
 
+type DeviceCodeCredentialOptions struct {
+	TenantID   *string                 // Gets the Azure Active Directory tenant (directory) ID of the service principal
+	ClientID   *string                 // Gets the client (application) ID of the service principal
+	UserPrompt *func(string)           // The callback function used to send the login message back to the user
```

Comment:
Funcs are implicitly `nil`able so no need for `*`.

Reply:
Done